packages/hardhat-core/test/internal/hardhat-network/console-log/test.ts

```diff
@@ -0,0 +1,73 @@
+import { assert } from "chai";
+import path from "path";
+import { Artifacts } from "../../../../src/internal/artifacts";
+import { useEnvironment } from "../../../helpers/environment";
+import { useFixtureProject } from "../../../helpers/project";
+import { TASK_COMPILE } from "../../../../src/builtin-tasks/task-names";
+
+const ARTIFACTS_FOLDER_PATH = path.join(
+  __dirname,
+  "..",
+  "..",
+  "..",
+  "fixture-projects",
+  "console-log",
+  "artifacts",
+  "contracts",
+  "Test.sol"
+);
+
+describe("Solidity console.log should print numbers without losing precision, occurrences of %d and %i should be correctly replaced with %s", function () {
+  // Set up the test environment
+  useFixtureProject("console-log");
+  useEnvironment();
+
+  it("should print all the numbers without losing precision", async function () {
+    await this.env.run(TASK_COMPILE, { quiet: true });
+
+    // Retrieve the artifact of the solidity file compiled in the previous command
+    const artifacts = new Artifacts(ARTIFACTS_FOLDER_PATH);
+    const artifact = artifacts.readArtifactSync("Test");
```

Comment:
You can drop the previous line and the `ARTIFACTS_FOLDER_PATH` variable, and to this instead:

```suggestion
    const artifact = this.env.artifacts.readArtifactSync("Test");
```

The environment is created with an `artifacts` field that is initialized with the right directory. Using this is both more concise and closer to what's going on in a real project.

Reply:
Done